runtime/expr/function/compare.go

```diff
@@ -0,0 +1,28 @@
+package function
+
+import (
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/runtime/expr"
+)
+
+func NewCompare() *Compare {
+	return &Compare{
+		fn: expr.NewValueCompareFn(true),
+	}
+}
+
+type Compare struct {
+	fn expr.CompareFn
+}
+
+func (e *Compare) Call(ctx zed.Allocator, args []zed.Value) *zed.Value {
+	lhs, rhs := &args[0], &args[1]
+	if lhs.IsError() || rhs.IsError() {
+		if lhs.IsError() {
+			return lhs
+		}
+		return rhs
+	}
+	cmp := e.fn(lhs, rhs)
+	return ctx.NewValue(zed.TypeInt64, zed.EncodeInt(int64(cmp)))
```

Comment:
1. I think we want to be able to compare error values.
2. Let `newInt64` allocate the `zcode.Bytes`. (It just calls `zed.EncodeInt` for now, but it'll be more efficient eventually.)
```suggestion
	return newInt64(ctx, int64(e.cmp(&args[0], &args[1]))
```